### PR TITLE
Add SSR check on CircleCI

### LINKED ISSUE #607: [FEATURE] Add SSR automatic check on CI
**Is your feature request related to a problem? Please describe.**
It can be easy to break the SSR feature of the kit by mistake in a PR, and hard to spot for small changes.
It would be useful to have an automatic check for each PR on CircleCI.

**Describe the solution you'd like**
Because there's already a SSR solution for this ( https://github.com/italia/design-italia-gatsby-starterkit ) It would be great to use it to check, something like:

```sh
$ npx gatsby new my-new-website https://github.com/italia/design-italia-gatsby-starterkit
$ cd my-new-website && npm build
```

### PR DESCRIPTION
**Pull request by dej611**
Fixes #607 

#### Short description of what this resolves:

This PR adds a new check to the CI by creating a new Gatsby project at each build using the curreent version of the toolkit.

#### Changes proposed in this pull request:

- 👷 Add new job for the SSR check

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -91,7 +91,37 @@ jobs:
...
+      - checkout
+
+      # Download and cache dependencies
+      - restore_cache: *restore_cache
+      - run: *npm_install
+      - save_cache: *save_cache
+
+      - run:
+          name: Change Permissions
+          command: sudo chown -R $(whoami) /usr/local
```
Comment (sebbalex) [inline .circleci/config.yml:109]: @dej611 do we really need this? This folder should be already accessible by user.

Comment (dej611) [reply .circleci/config.yml:109]: I tried without it [here](https://app.circleci.com/pipelines/github/italia/design-react-kit/161/workflows/85bae6d4-2365-40fc-84f7-41f145c3d5ae/jobs/2877) and it raises permission problems. Looking on Internet I found this solution. I'm open to alternatives that works :D

Comment (sebbalex) [reply .circleci/config.yml:109]:
```suggestion
          name: Bind global package folder to local user
          command: |
              mkdir ~/.npm-packages
              npm config set prefix ~/.npm-packages
```
This should do the trick. We will use a local user folder to store all global packages. What do you think?

**Review by sebbalex — APPROVED** [review]
:rocket:

**Comment by dej611** [comment]
Now it builds the right thing 🎉

**Review by sebbalex — APPROVED** [review]

**Pull request merged**